hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/fs/AbstractSpaceUsageSource.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hdds.fs;
+
+import org.apache.hadoop.util.Time;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.function.LongSupplier;
+
+/**
+ * Convenience parent class for {@code SpaceUsageSource} implementations.
+ */
+public abstract class AbstractSpaceUsageSource implements SpaceUsageSource {
+
+  private final File file;
+  private final String path;
+
+  /**
+   * @param file the path to check disk usage in
+   */
+  protected AbstractSpaceUsageSource(File file) {
+    this.file = file;
+    try {
+      this.path = file.getCanonicalPath();
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  protected static long time(LongSupplier supplier, Logger logger) {
```

Comment:
Could you add a short javadoc? I got what it does but not clear from first reading.

Reply:
Good idea, I had to re-read it, too. ;)